File: software/examples/vbnet/ExampleThreshold.vb
Imports Tinkerforge

Module ExampleThreshold
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change to your UID

    ' Callback for illuminance greater than 200 Lux
    Sub ReachedCB(ByVal sender As BrickletAmbientLightV2, ByVal illuminance As Long)
        System.Console.WriteLine("We have " + (illuminance/100.0).ToString() + " Lux.")
        System.Console.WriteLine("Too bright, close the curtains!")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim al As New BrickletAmbientLightV2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        al.SetDebouncePeriod(10000)

        ' Register threshold reached callback to function ReachedCB
        AddHandler al.IlluminanceReached, AddressOf ReachedCB

        ' Configure threshold for "greater than 200 Lux" (unit is Lux/100)
        al.SetIlluminanceCallbackThreshold(">"C, 200*100, 0)

        System.Console.WriteLine("Press key to exit")
        System.Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/java/ExampleThreshold.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletAmbientLightV2;

public class ExampleThreshold {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;
	private static final String UID = "XYZ"; // Change to your UID

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletAmbientLightV2 al = new BrickletAmbientLightV2(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 10 seconds (10000ms)
		al.setDebouncePeriod(10000);

		// Add illuminance reached listener (parameter has unit Lux/100)
		al.addIlluminanceReachedListener(new BrickletAmbientLightV2.IlluminanceReachedListener() {
			public void illuminanceReached(long illuminance) {
				System.out.println("Illuminance: " + illuminance/100.0 + " Lux");
				System.out.println("Too bright, close the curtains!");
			}
		});

		// Configure threshold for illuminance "greater than 500 Lux" (unit is Lux/100)
		al.setIlluminanceCallbackThreshold('>', 500*100, 0);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
